<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>API &#8212; dms2dfe 1.0.6 documentation</title>
    <link rel="stylesheet" href="../../_static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../',
        VERSION:     '1.0.6',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="../../_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="../../_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="../../_static/bootstrap-3.3.7/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="../../_static/bootstrap-sphinx.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">

  </head>
  <body>

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="../../index.html">
          dms2dfe</a>
        <span class="navbar-text navbar-version pull-left"><b>1.0.6</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
                <li><a href="../../1installation.html">Installation</a></li>
                <li><a href="../../2usage.html">Usage</a></li>
                <li><a href="../../6test.html">Example</a></li>
                <li><a href="../../4io.html">I/O</a></li>
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="../../index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul>
<li class="toctree-l1"><a class="reference internal" href="../../1installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../2usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../3programs.html">Programs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../4io.html">I/O</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../6test.html">Example Datasets</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../5api.html">API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../1installation_deps.html">Dependencies</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../7troubleshoot.html">Troubleshoot</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../6contributors.html">Guidelines for contributors</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../8future.html">Future releases</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">API</a><ul>
<li><a class="reference internal" href="#convert-seq"><code class="docutils literal"><span class="pre">convert_seq</span></code></a></li>
<li><a class="reference internal" href="#fit-curve"><code class="docutils literal"><span class="pre">fit_curve</span></code></a></li>
<li><a class="reference internal" href="#get-protein-features"><code class="docutils literal"><span class="pre">get_protein_features</span></code></a></li>
<li><a class="reference internal" href="#global-vars"><code class="docutils literal"><span class="pre">global_vars</span></code></a></li>
<li><a class="reference internal" href="#io-seq-files"><code class="docutils literal"><span class="pre">io_seq_files</span></code></a></li>
<li><a class="reference internal" href="#io-data-files"><code class="docutils literal"><span class="pre">io_data_files</span></code></a></li>
<li><a class="reference internal" href="#io-ml"><code class="docutils literal"><span class="pre">io_ml</span></code></a></li>
<li><a class="reference internal" href="#io-mut-files"><code class="docutils literal"><span class="pre">io_mut_files</span></code></a></li>
<li><a class="reference internal" href="#plot-pdb"><code class="docutils literal"><span class="pre">plot_pdb</span></code></a></li>
<li><a class="reference internal" href="#id8"><code class="docutils literal"><span class="pre">io_data_files</span></code></a></li>
<li><a class="reference internal" href="#plot-mut-data"><code class="docutils literal"><span class="pre">plot_mut_data</span></code></a></li>
<li><a class="reference internal" href="#io-nums"><code class="docutils literal"><span class="pre">io_nums</span></code></a></li>
<li><a class="reference internal" href="#io-strs"><code class="docutils literal"><span class="pre">io_strs</span></code></a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
            
            
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="../../search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
    <div class="col-md-12 content">
      
  <div class="section" id="api">
<span id="id1"></span><h1>API<a class="headerlink" href="#api" title="Permalink to this headline">¶</a></h1>
<p>Backend modules of <a class="reference internal" href="index.html#dms2dfe"><span class="std std-ref">dms2dfe</span></a>.</p>
<span class="target" id="module-dms2dfe.lib.convert_seq"></span><div class="section" id="convert-seq">
<h2><code class="docutils literal"><span class="pre">convert_seq</span></code><a class="headerlink" href="#convert-seq" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.convert_seq.cds2aas">
<code class="descclassname">dms2dfe.lib.convert_seq.</code><code class="descname">cds2aas</code><span class="sig-paren">(</span><em>cds_str</em>, <em>host</em>, <em>stop_codon='X'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.convert_seq.cds2aas" title="Permalink to this definition">¶</a></dt>
<dd><p>This translates codons to amino acids.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>cds_str</strong> – codon sequence string.</li>
<li><strong>host</strong> – name of host organism for choosing codon table. [coli | yeast | sapiens].</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns aas:</th><td class="field-body"><p class="first last">amino acid.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.convert_seq.cmplmt">
<code class="descclassname">dms2dfe.lib.convert_seq.</code><code class="descname">cmplmt</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.convert_seq.cmplmt" title="Permalink to this definition">¶</a></dt>
<dd><blockquote>
<div>Gets complement of a sequence.</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>s</strong> – sequence string.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.convert_seq.revcom">
<code class="descclassname">dms2dfe.lib.convert_seq.</code><code class="descname">revcom</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.convert_seq.revcom" title="Permalink to this definition">¶</a></dt>
<dd><blockquote>
<div>Gets reverse complement of a sequence.</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>s</strong> – sequence string.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.convert_seq.revers">
<code class="descclassname">dms2dfe.lib.convert_seq.</code><code class="descname">revers</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.convert_seq.revers" title="Permalink to this definition">¶</a></dt>
<dd><blockquote>
<div>Gets reverse of a sequence.</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>s</strong> – sequence string.</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.fit_curve"></span><div class="section" id="fit-curve">
<h2><code class="docutils literal"><span class="pre">fit_curve</span></code><a class="headerlink" href="#fit-curve" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.fit_curve.fit_gauss">
<code class="descclassname">dms2dfe.lib.fit_curve.</code><code class="descname">fit_gauss</code><span class="sig-paren">(</span><em>x</em>, <em>a</em>, <em>x0</em>, <em>sigma</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.fit_curve.fit_gauss" title="Permalink to this definition">¶</a></dt>
<dd><p>This fits gaussian.
Eq: a*np.exp(-(x-x0)**2/(2*sigma**2))</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>x</strong> – value of x</li>
<li><strong>a</strong> – value of a</li>
<li><strong>x0</strong> – value of x0</li>
<li><strong>sigma</strong> – value of sigma</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.fit_curve.fit_gauss_params">
<code class="descclassname">dms2dfe.lib.fit_curve.</code><code class="descname">fit_gauss_params</code><span class="sig-paren">(</span><em>data_vector</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.fit_curve.fit_gauss_params" title="Permalink to this definition">¶</a></dt>
<dd><p>This function is used to calculate the mximum likelihood of Fitness of synonymous mutations (later used for normalization).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data_vector</strong> – pandas column.</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.get_protein_features"></span><div class="section" id="get-protein-features">
<h2><code class="docutils literal"><span class="pre">get_protein_features</span></code><a class="headerlink" href="#get-protein-features" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.concat_feats">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">concat_feats</code><span class="sig-paren">(</span><em>data_feats_all</em>, <em>data_feats</em>, <em>col_index</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.concat_feats" title="Permalink to this definition">¶</a></dt>
<dd><p>Concatenates tables containing individual mutation wise and position wise features.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_feats_all</strong> – mutation wise features</li>
<li><strong>data_feats</strong> – position wise features</li>
<li><strong>col_index</strong> – columns to be concatenated</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_bfactor">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_bfactor</code><span class="sig-paren">(</span><em>pdb_fh</em>, <em>ref_out=False</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_bfactor" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts B (temperature) factor from PDB structure</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>pdb_fh</strong> – path to PDB structure file</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_bfactor:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body">pandas table with B factor per residue</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_consrv_score">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_consrv_score</code><span class="sig-paren">(</span><em>fsta_fh</em>, <em>host</em>, <em>clustalo_fh</em>, <em>rate4site_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_consrv_score" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts Residue wise conservation scores</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>fsta_fh</strong> – path to FASTA file</li>
<li><strong>host</strong> – host name e.g. E. coli</li>
<li><strong>clustalo_fh</strong> – path to Clustalo libraries</li>
<li><strong>rate4site_fh</strong> – path to rate4site libraries</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_conserv:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">pandas table with conservation scores per residue</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_data_feats_all">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_data_feats_all</code><span class="sig-paren">(</span><em>data_feats_mut_fh</em>, <em>data_feats_pos_fh</em>, <em>data_feats_sub_fh</em>, <em>data_out_fh</em>, <em>info</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_data_feats_all" title="Permalink to this definition">¶</a></dt>
<dd><p>Combines all the extracted features.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>data_feats_mut_fh</strong> – path to file containing individual mutation-wise features</li>
<li><strong>data_feats_pos_fh</strong> – path to file containing position-wise features</li>
<li><strong>data_feats_sub_fh</strong> – path to file containing substitution-wise features</li>
<li><strong>data_out_fh</strong> – path to the file where the combined data would be saved</li>
<li><strong>info</strong> – Information of the experiment, taken from prj_dh/cfg directory</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_feats_all:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">pandas table with combined data.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_data_feats_mut">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_data_feats_mut</code><span class="sig-paren">(</span><em>prj_dh</em>, <em>data_out_fh</em>, <em>info</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_data_feats_mut" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts individual mutation wise features</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>prj_dh</strong> – project directory</li>
<li><strong>data_out_fh</strong> – path to output table</li>
<li><strong>info</strong> – dict, information about the project</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_feats_mut:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">pandas table with mutation wise features</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_data_feats_mut_diffs">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_data_feats_mut_diffs</code><span class="sig-paren">(</span><em>data_feats_pos</em>, <em>feats_sub_pos_tups</em>, <em>data_out_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_data_feats_mut_diffs" title="Permalink to this definition">¶</a></dt>
<dd><p>This calculates arithmatic operations (+-/<a href="#id2"><span class="problematic" id="id3">*</span></a>) on the selected features.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>data_feats_pos</strong> – pandas table of position wise features.</li>
<li><strong>feats_sub_pos</strong> – pandas table of substitution wise features.</li>
<li><strong>data_ous_fh</strong> – path to output table.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns dfromligands:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">pandas table with distances from ligand</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_data_feats_pos">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_data_feats_pos</code><span class="sig-paren">(</span><em>prj_dh</em>, <em>info</em>, <em>data_out_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_data_feats_pos" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts position wise features</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>prj_dh</strong> – path to project directory</li>
<li><strong>data_out_fh</strong> – path to output table</li>
<li><strong>info</strong> – dict, information about the project</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_feats_pos:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">pandas table with position wise features</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_data_feats_sub">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_data_feats_sub</code><span class="sig-paren">(</span><em>data_out_fh</em>, <em>data_feats_aas_fh='/home/anonymous/Documents/propro/writ/prjs/1_dms_software/progs/dms2dfe/dms2dfe/lib/data_feats_aas'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_data_feats_sub" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts substitution wise features</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>data_out_fh</strong> – path to output table</li>
<li><strong>data_feats_aas_fh</strong> – path to extracted input features</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_feats_sub:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">pandas table with substitution wise features</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_dfrominterface">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_dfrominterface</code><span class="sig-paren">(</span><em>pdb_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_dfrominterface" title="Permalink to this definition">¶</a></dt>
<dd><p>This calculates distances between each ligand atom or optionally provided amino acids (sources) and each residue in the protein.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>pdb_fh</strong> – path to .pdb file.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns dinter:</th><td class="field-body">pandas table with distances from dimer interface</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_pdb_feats">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_pdb_feats</code><span class="sig-paren">(</span><em>pdb_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_pdb_feats" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts features from PDB structure</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>pdb_fh</strong> – path to PDB structure file</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns pdb_feats:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body">pandas table with PDB features</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.get_residue_depth">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">get_residue_depth</code><span class="sig-paren">(</span><em>pdb_fh</em>, <em>msms_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.get_residue_depth" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts Residue depth from PDB structure</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>pdb_fh</strong> – path to PDB structure file</li>
<li><strong>msms_fh</strong> – path to MSMS libraries</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_depth:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">pandas table with residue depth per residue</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.getdssp_data">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">getdssp_data</code><span class="sig-paren">(</span><em>pdb_fh</em>, <em>dssp_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.getdssp_data" title="Permalink to this definition">¶</a></dt>
<dd><p>This uses DSSP to get structural information.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>pdb_fh</strong> – path to .pdb file</li>
<li><strong>dssp_fh</strong> – path to DSSP source.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns dssp_data:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">pandas table with extracted DSSP data</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.get_protein_features.pdb2dfromactivesite">
<code class="descclassname">dms2dfe.lib.get_protein_features.</code><code class="descname">pdb2dfromactivesite</code><span class="sig-paren">(</span><em>pdb_fh</em>, <em>active_sites=[]</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.get_protein_features.pdb2dfromactivesite" title="Permalink to this definition">¶</a></dt>
<dd><p>This calculates distances between each ligand atom or optionally provided amino acids (sources) and each residue in the protein.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>pdb_fh</strong> – path to .pdb file.</li>
<li><strong>active_sites</strong> – optional list of residue numbers as sources.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns dfromligands:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">pandas table with distances from ligand</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.global_vars"></span><div class="section" id="global-vars">
<h2><code class="docutils literal"><span class="pre">global_vars</span></code><a class="headerlink" href="#global-vars" title="Permalink to this headline">¶</a></h2>
<p>This contains all the global variables used variously in the package.</p>
</div>
<span class="target" id="module-dms2dfe.lib.io_seq_files"></span><div class="section" id="io-seq-files">
<h2><code class="docutils literal"><span class="pre">io_seq_files</span></code><a class="headerlink" href="#io-seq-files" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.Rs2mergedandjoined">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">Rs2mergedandjoined</code><span class="sig-paren">(</span><em>fastq_R1_read</em>, <em>fastq_R2_read</em>, <em>fastq_merged_f</em>, <em>fastq_joined_f</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.Rs2mergedandjoined" title="Permalink to this definition">¶</a></dt>
<dd><p>This merges and joins R1 and R2 reads.</p>
<p>merged files (<a href="#id4"><span class="problematic" id="id5">*</span></a>_dplxd_merged.fastq) have the R1 and R2 mixed together without joining. 
eg. R1 R2 R1 R2 ..</p>
<p>merged files (<a href="#id6"><span class="problematic" id="id7">*</span></a>_dplxd_joined.fastq) have the R1 and R2 joined together.
eg.    R1R2 R1R2 ..</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fastq_R1_read</strong> – sequence of R1 read of .fastq (str).</li>
<li><strong>fastq_R2_read</strong> – sequence of R2 read of .fastq (str).</li>
<li><strong>fastq_merged_f</strong> – file object for output merged .fastq.</li>
<li><strong>fastq_joined_f</strong> – file object for output joined .fastq.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.cctmr_fasta2ref_fasta">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">cctmr_fasta2ref_fasta</code><span class="sig-paren">(</span><em>fsta_fh</em>, <em>cctmr</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.cctmr_fasta2ref_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts concatamer sequence to monomer fasta.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fsta_fh</strong> – path to fasta file</li>
<li><strong>cctmr</strong> – seqeunce of individual monoer sequence</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.contains_barcode">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">contains_barcode</code><span class="sig-paren">(</span><em>barcode_seq</em>, <em>read_seq</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.contains_barcode" title="Permalink to this definition">¶</a></dt>
<dd><p>This searches for barcode sequence in given sequence read     in windows of +/- 5 nucleotide upstream and downstream.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>barcode_seq</strong> – sequence of barcode (string).</li>
<li><strong>barcode_seq</strong> – target sequence (string).</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.fasta_nts2prt">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">fasta_nts2prt</code><span class="sig-paren">(</span><em>fsta_fh</em>, <em>host='coli'</em>, <em>fsta_prt_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.fasta_nts2prt" title="Permalink to this definition">¶</a></dt>
<dd><p>Translates nucleotide fasta to amino acid fasta</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>fsta_fh</strong> – path to fasta file</li>
<li><strong>host</strong> – host organism e.g. E. coli</li>
<li><strong>fsta_prt_fh</strong> – path to fasta protein sequence</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns fsta_seq_prt:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">fasta protein sequence</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.fasta_writer">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">fasta_writer</code><span class="sig-paren">(</span><em>otpt_f</em>, <em>read_id</em>, <em>read_seq</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.fasta_writer" title="Permalink to this definition">¶</a></dt>
<dd><p>This appends a sequence reads in fastq format.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>otpt_f</strong> – file object of output file.</li>
<li><strong>read_id</strong> – ID of fastq file (str).</li>
<li><strong>read_seq</strong> – sequence (str).</li>
<li><strong>read_qua</strong> – quality scores (str).</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.fastq2dplx">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">fastq2dplx</code><span class="sig-paren">(</span><em>fastq_R1_fh</em>, <em>fastq_R2_fh</em>, <em>barcode_R1s</em>, <em>barcode_R2s</em>, <em>fastq_fns</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.fastq2dplx" title="Permalink to this definition">¶</a></dt>
<dd><p>This demultiplexes fastq.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fastq_R1_fh</strong> – path to R1 .fastq file.</li>
<li><strong>fastq_R2_fh</strong> – path to R2 .fastq file.</li>
<li><strong>barcode_R1s</strong> – barcodes to demultiplex forward reads.</li>
<li><strong>barcode_R1s</strong> – barcodes to demultiplex reverse reads.</li>
<li><strong>fastq_fns</strong> – file names of demultiplexed .fastq files (saved in same folder as the non-multiplexed input file).</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.fastq2qcd">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">fastq2qcd</code><span class="sig-paren">(</span><em>fastq_fhs_list</em>, <em>trimmomatic_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.fastq2qcd" title="Permalink to this definition">¶</a></dt>
<dd><p>Uses Trimmomatic to filter and trim low quality reads.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fastq_fhs_list</strong> – list of input .fastq files.</li>
<li><strong>trimmomatic_fh</strong> – path to trimmomatic source file.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.fastq_writer">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">fastq_writer</code><span class="sig-paren">(</span><em>otpt_f</em>, <em>read_id</em>, <em>read_seq</em>, <em>read_qua</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.fastq_writer" title="Permalink to this definition">¶</a></dt>
<dd><p>This appends a sequence reads in fastq format.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>otpt_f</strong> – file object of output file.</li>
<li><strong>read_id</strong> – ID of fastq file (str).</li>
<li><strong>read_seq</strong> – sequence (str).</li>
<li><strong>read_qua</strong> – quality scores (str).</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.get_fsta_feats">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">get_fsta_feats</code><span class="sig-paren">(</span><em>fsta_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.get_fsta_feats" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets basic fasta features e.g. length name etc</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>fsta_fh</strong> – path to fasta file</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns fsta_id,fsta_seq,fsta_len:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body">id, sequence, length</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.getcov">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">getcov</code><span class="sig-paren">(</span><em>sbam_fh</em>, <em>fsta_fh</em>, <em>refini=0</em>, <em>refend=None</em>, <em>data_out_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.getcov" title="Permalink to this definition">¶</a></dt>
<dd><p>Get coverage statistics from sorted bam file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>sbam_fh</strong> – path to sorted bam file</li>
<li><strong>fsta_fh</strong> – path to fasta file</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Pram refini:</th><td class="field-body"><p class="first">int, start index</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Pram refend:</th><td class="field-body"><p class="first">int, stop index</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Pram data_out_fh:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">string, path to output table</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.getdepth_cds">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">getdepth_cds</code><span class="sig-paren">(</span><em>sbam_fh</em>, <em>fsta_fh</em>, <em>cctmr=None</em>, <em>refini=0</em>, <em>refend=None</em>, <em>data_out_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.getdepth_cds" title="Permalink to this definition">¶</a></dt>
<dd><p>get codon level depth from sorted bam file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>sbam_fh</strong> – path to sorted bam file</li>
<li><strong>fsta_fh</strong> – path to fasta file</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Pram refini:</th><td class="field-body"><p class="first">int, start index</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Pram refend:</th><td class="field-body"><p class="first">int, stop index</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Pram data_out_fh:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">string, path to output table</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.getdepth_ref">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">getdepth_ref</code><span class="sig-paren">(</span><em>sbam_fh</em>, <em>fsta_fh</em>, <em>cctmr=None</em>, <em>refini=0</em>, <em>refend=None</em>, <em>data_out_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.getdepth_ref" title="Permalink to this definition">¶</a></dt>
<dd><p>This gets the codon level coverage of .fastq files.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>lbl</strong> – name of sample.</li>
<li><strong>lbls</strong> – dataframe with configuration <cite>lbls</cite>.</li>
<li><strong>cctmr</strong> – tuple with (1-based) boundaries of concatamers.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns wt:</th><td class="field-body"><p class="first last">dataframe with wild-type coverage.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.getusablefastqs_list">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">getusablefastqs_list</code><span class="sig-paren">(</span><em>prj_dh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.getusablefastqs_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks sanity of fastq files mentioned in configuration excel file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>prj_dh</strong> – path to project directory.</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns fastq_fhs:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body">list of paths to .fastq files chich can be used downstream.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_seq_files.qcd2sbam">
<code class="descclassname">dms2dfe.lib.io_seq_files.</code><code class="descname">qcd2sbam</code><span class="sig-paren">(</span><em>fastq_fhs_list</em>, <em>fsta_fh</em>, <em>alignment_type</em>, <em>bt2_ref_fh</em>, <em>bowtie2_fh</em>, <em>samtools_fh</em>, <em>bowtie2_com=''</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_seq_files.qcd2sbam" title="Permalink to this definition">¶</a></dt>
<dd><p>This aligns the fastq files using bowtie2.
The resultant .sam files are converted to sorted .bam files (.sbam) for downstream ana1_sam2mutmat module.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fastq_fhs_list</strong> – list of input .fastq files.</li>
<li><strong>fsta_fh</strong> – path to reference .fasta file.</li>
<li><strong>alignment_type</strong> – type of alignment [‘loc’ : local | ‘glob’ : global]</li>
<li><strong>bt2_ref_fh</strong> – path to bowtie2 reference file.</li>
<li><strong>bowtie2_fh</strong> – path to bowtie2 source file.</li>
<li><strong>samtools_fh</strong> – path to samtools source file.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.io_data_files"></span><div class="section" id="io-data-files">
<h2><code class="docutils literal"><span class="pre">io_data_files</span></code><a class="headerlink" href="#io-data-files" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.io_data_files.auto_find_missing_paths">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">auto_find_missing_paths</code><span class="sig-paren">(</span><em>prj_dh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.auto_find_missing_paths" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds the missing paths in the configuration given in cfg/ directory</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>prj_dh</strong> – path to the project directory</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.convert2h5form">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">convert2h5form</code><span class="sig-paren">(</span><em>df</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.convert2h5form" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert dataframe compatible to  Hdf5 format</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>df</strong> – pandas dataframe</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.csv2src">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">csv2src</code><span class="sig-paren">(</span><em>csv_fh</em>, <em>src_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.csv2src" title="Permalink to this definition">¶</a></dt>
<dd><p>This writes <cite>.csv</cite> to <cite>.py</cite> source file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>csv_fh</strong> – path to input <cite>.csv</cite> file.</li>
<li><strong>src_fh</strong> – path to output <cite>.py</cite> source file.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.csvs2h5">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">csvs2h5</code><span class="sig-paren">(</span><em>dh</em>, <em>sub_dh_list</em>, <em>fn_list</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.csvs2h5" title="Permalink to this definition">¶</a></dt>
<dd><p>This converts csvs into HDF5 tables.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>dh</strong> – path to the directory with csv files</li>
<li><strong>fn_list</strong> – list of filenames of the csv files</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.get_raw_input">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">get_raw_input</code><span class="sig-paren">(</span><em>info</em>, <em>var</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.get_raw_input" title="Permalink to this definition">¶</a></dt>
<dd><p>Get intearactive inputs from user</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>info</strong> – dict, with information about experiment</li>
<li><strong>var</strong> – variable whose value is obtained from interactive shell</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.getusable_comparison_list">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">getusable_comparison_list</code><span class="sig-paren">(</span><em>prj_dh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.getusable_comparison_list" title="Permalink to this definition">¶</a></dt>
<dd><p>This converts the table of tests and controls in configuration file into tuples of test and control.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>prj_dh</strong> – path to project directory.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.getusable_fits_list">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">getusable_fits_list</code><span class="sig-paren">(</span><em>prj_dh</em>, <em>data_fit_dh='data_fit'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.getusable_fits_list" title="Permalink to this definition">¶</a></dt>
<dd><p>This gets the list of samples that can be processed for fitness estimations.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>prj_dh</strong> – path to project directory.</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns fits_pairs_list:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body">list of tuples with names of input and selected samples.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.getusable_lbls_list">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">getusable_lbls_list</code><span class="sig-paren">(</span><em>prj_dh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.getusable_lbls_list" title="Permalink to this definition">¶</a></dt>
<dd><p>This detects the samples that can be processed.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>prj_dh</strong> – path to project directory.</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns lbls_list:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body">list of names of samples that can be processed.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.info2src">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">info2src</code><span class="sig-paren">(</span><em>prj_dh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.info2src" title="Permalink to this definition">¶</a></dt>
<dd><p>This converts <cite>.csv</cite> configuration file to <cite>.py</cite> source file saved in <cite>/tmp/</cite>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>prj_dh</strong> – path to project directory</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.is_cfg_ok">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">is_cfg_ok</code><span class="sig-paren">(</span><em>cfg_dh</em>, <em>cfgs</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.is_cfg_ok" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if the required files are present in given directory.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>cfg_dh</strong> – path to directory.</li>
<li><strong>cfgs</strong> – list of names of files.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.is_info_ok">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">is_info_ok</code><span class="sig-paren">(</span><em>xls_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.is_info_ok" title="Permalink to this definition">¶</a></dt>
<dd><p>This checks the sanity of info sheet in the configuration excel file.
For example if the files exists or not.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>cfg_xls</strong> – path to configuration excel file</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.is_xls_ok">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">is_xls_ok</code><span class="sig-paren">(</span><em>cfg_xls</em>, <em>cfg_xls_sheetnames_required</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.is_xls_ok" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks if the required sheets are present in the configuration excel file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>cfg_xls</strong> – path to configuration excel file</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.raw_input2info">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">raw_input2info</code><span class="sig-paren">(</span><em>prj_dh</em>, <em>inputORdefault</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.raw_input2info" title="Permalink to this definition">¶</a></dt>
<dd><p>This writes configuration <cite>.csv</cite> file from <cite>raw_input</cite> from prompt.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>prj_dh</strong> – path to project directory.</li>
<li><strong>inputORdefault</strong> – column name “input” or “default”.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.read_pkl">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">read_pkl</code><span class="sig-paren">(</span><em>fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.read_pkl" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads a file in pkl format</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>fh</strong> – path to the pkl file</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns data:</th><td class="field-body">dict, containing data</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.to_pkl">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">to_pkl</code><span class="sig-paren">(</span><em>data</em>, <em>fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.to_pkl" title="Permalink to this definition">¶</a></dt>
<dd><p>Saves a dict in pkl format</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data</strong> – dict, containing data</li>
<li><strong>fh</strong> – path to the output pkl file</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.xls2csvs">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">xls2csvs</code><span class="sig-paren">(</span><em>cfg_xls</em>, <em>cfg_xls_sheetnames_required</em>, <em>output_dh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.xls2csvs" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts configuration excel file to HDF5(h5) file.
Here sheets in excel files are converted to groups in HDF5 file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>cfg_xls</strong> – path to configuration excel file</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_data_files.xls2h5">
<code class="descclassname">dms2dfe.lib.io_data_files.</code><code class="descname">xls2h5</code><span class="sig-paren">(</span><em>cfg_xls</em>, <em>cfg_h5</em>, <em>cfg_xls_sheetnames_required</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_data_files.xls2h5" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts configuration excel file to HDF5(h5) file.
Here sheets in excel files are converted to groups in HDF5 file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>cfg_xls</strong> – path to configuration excel file</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.io_ml"></span><div class="section" id="io-ml">
<h2><code class="docutils literal"><span class="pre">io_ml</span></code><a class="headerlink" href="#io-ml" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.io_ml.corrplot">
<code class="descclassname">dms2dfe.lib.io_ml.</code><code class="descname">corrplot</code><span class="sig-paren">(</span><em>info</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_ml.corrplot" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots a correlation matrix heatmap between range of features and fold change values</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>info</strong> – dict, with the information of the experiment</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_ml.dXy2ml">
<code class="descclassname">dms2dfe.lib.io_ml.</code><code class="descname">dXy2ml</code><span class="sig-paren">(</span><em>dXy</em>, <em>ycol</em>, <em>params=None</em>, <em>if_gridsearch=False</em>, <em>if_partial_dependence=False</em>, <em>if_feats_imps=False</em>, <em>inter=None</em>, <em>use_top=None</em>, <em>out_fh=None</em>, <em>regORcls='reg'</em>, <em>force=False</em>, <em>cores=8</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_ml.dXy2ml" title="Permalink to this definition">¶</a></dt>
<dd><p>Wrapper for ml operations</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>dXy</strong> – pandas table with preditors (X) and target (y) values</li>
<li><strong>ycol</strong> – column name of the target column</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_ml.data_combo2ml">
<code class="descclassname">dms2dfe.lib.io_ml.</code><code class="descname">data_combo2ml</code><span class="sig-paren">(</span><em>data_combo</em>, <em>data_fn</em>, <em>data_dh</em>, <em>plot_dh</em>, <em>ycoln</em>, <em>col_idx</em>, <em>ml_type='both'</em>, <em>middle_percentile_skipped=0.1</em>, <em>force=False</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_ml.data_combo2ml" title="Permalink to this definition">¶</a></dt>
<dd><p>This runs the submodules to run classifier from fitness data (<cite>data_combo</cite>).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>basename</strong><strong>(</strong><strong>data_fn</strong><strong>)</strong> – in the form &lt;data_combo&gt;/&lt;aas/cds&gt;/&lt;name of file&gt;.</li>
<li><strong>data_feats</strong> – dataframe with features.</li>
<li><strong>y_coln</strong> – column name of column with classes (ys).</li>
<li><strong>ml_type</strong> – classi | both</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_ml.data_fit2ml">
<code class="descclassname">dms2dfe.lib.io_ml.</code><code class="descname">data_fit2ml</code><span class="sig-paren">(</span><em>dX_fh</em>, <em>dy_fh</em>, <em>info</em>, <em>regORcls='cls'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_ml.data_fit2ml" title="Permalink to this definition">¶</a></dt>
<dd><p>Wrapper for overall data_fit to regression modelling</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>dX_fh</strong> – path to the file containing preditor values</li>
<li><strong>dy_fh</strong> – path to the file containing target values</li>
<li><strong>info</strong> – dict contaning information about the experiment</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_ml.data_regress2data_fit">
<code class="descclassname">dms2dfe.lib.io_ml.</code><code class="descname">data_regress2data_fit</code><span class="sig-paren">(</span><em>prj_dh</em>, <em>data_fit_key</em>, <em>data_regress_all</em>, <em>col='FCA_norm'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_ml.data_regress2data_fit" title="Permalink to this definition">¶</a></dt>
<dd><p>Transforms the fold changes estimated from a regression model in the format of data_fit</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>prj_dh</strong> – path to the project dorectory</li>
<li><strong>data_fit_key</strong> – path key to data_fit file</li>
<li><strong>data_regress_all</strong> – pandas table with regression estimated fold change values</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_ml.est2feats_imp">
<code class="descclassname">dms2dfe.lib.io_ml.</code><code class="descname">est2feats_imp</code><span class="sig-paren">(</span><em>est</em>, <em>Xcols</em>, <em>Xy=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_ml.est2feats_imp" title="Permalink to this definition">¶</a></dt>
<dd><p>Get Feature importances from estimator</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>est</strong> – Estimator object</li>
<li><strong>Xcols</strong> – list of column names of predictors</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_ml.run_RF_classi">
<code class="descclassname">dms2dfe.lib.io_ml.</code><code class="descname">run_RF_classi</code><span class="sig-paren">(</span><em>data_all</em>, <em>X_cols</em>, <em>y_coln</em>, <em>test_size=0.34</em>, <em>data_test=None</em>, <em>data_out_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_ml.run_RF_classi" title="Permalink to this definition">¶</a></dt>
<dd><p>This implements Random Forest classifier.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>data_all</strong> – dataframe with columns with features(Xs) and classes(y).</li>
<li><strong>X_cols</strong> – list of column names with features.</li>
<li><strong>y_coln</strong> – column name of column with classes.</li>
<li><strong>plot_fh</strong> – path to output plot file.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns grid_search:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first">trained classifier object.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns y_test:</th><td class="field-body"><p class="first">classes used for testing classifier.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns y_pred:</th><td class="field-body"><p class="first">predicted classes.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Returns y_score:</th></tr>
<tr class="field-odd field"><td>&#160;</td><td class="field-body"><p class="first">scores of predicted classes used to plot ROC curve.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns feature_importances:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">relative importances of features (dataframe).</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_ml.run_RF_regress">
<code class="descclassname">dms2dfe.lib.io_ml.</code><code class="descname">run_RF_regress</code><span class="sig-paren">(</span><em>data_all</em>, <em>X_cols</em>, <em>y_coln</em>, <em>test_size=0.5</em>, <em>data_test=None</em>, <em>data_out_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_ml.run_RF_regress" title="Permalink to this definition">¶</a></dt>
<dd><p>This implements Random Forest classifier.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>data_all</strong> – dataframe with columns with features(Xs) and classes(y).</li>
<li><strong>X_cols</strong> – list of column names with features.</li>
<li><strong>y_coln</strong> – column name of column with classes.</li>
<li><strong>plot_fh</strong> – path to output plot file.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns grid_search:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first">trained classifier object.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns y_test:</th><td class="field-body"><p class="first">classes used for testing classifier.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns y_pred:</th><td class="field-body"><p class="first">predicted classes.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Returns y_score:</th></tr>
<tr class="field-odd field"><td>&#160;</td><td class="field-body"><p class="first">scores of predicted classes used to plot ROC curve.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns feature_importances:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">relative importances of features (dataframe).</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_ml.run_est">
<code class="descclassname">dms2dfe.lib.io_ml.</code><code class="descname">run_est</code><span class="sig-paren">(</span><em>est</em>, <em>X</em>, <em>y</em>, <em>params</em>, <em>cv=True</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_ml.run_est" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs an estimator</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>est</strong> – estimator object</li>
<li><strong>X</strong> – predictors (X) values</li>
<li><strong>y</strong> – target (y) values</li>
<li><strong>params</strong> – additional fitting parameters</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.io_mut_files"></span><div class="section" id="io-mut-files">
<h2><code class="docutils literal"><span class="pre">io_mut_files</span></code><a class="headerlink" href="#io-mut-files" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.collate_cctmr">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">collate_cctmr</code><span class="sig-paren">(</span><em>lbl_mat_cds</em>, <em>cctmr</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.collate_cctmr" title="Permalink to this definition">¶</a></dt>
<dd><p>If the reference sequence is a concatamer, this function collates the repeating mutation matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>lbl_mat_cds</strong> – codon level mutation matrix.</li>
<li><strong>cctmr</strong> – tuple with (1-based) boundaries of concatamers.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns lbl_mat_cds:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">collated codon level mutation matrix.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.data_fit2data_comparison">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">data_fit2data_comparison</code><span class="sig-paren">(</span><em>lbl_ctrl</em>, <em>lbl_test</em>, <em>prj_dh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.data_fit2data_comparison" title="Permalink to this definition">¶</a></dt>
<dd><p>This converts the Fitness values to Relative fitness among test and control fed as input.</p>
<div class="highlight-text"><div class="highlight"><pre><span></span>class fit:           beneficial, neutral, deleterious
class comparison:    killed, survived
class comparison:    positive, negative, robust
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>lbl_ctrl</strong> – name of control sample.</li>
<li><strong>lbl_test</strong> – name of test sample.</li>
<li><strong>prj_dh</strong> – path to project directory.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.data_lbl2data_fit">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">data_lbl2data_fit</code><span class="sig-paren">(</span><em>data_lbl_ref_fn, data_lbl_sel_fn, info, type_forms=['aas']</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.data_lbl2data_fit" title="Permalink to this definition">¶</a></dt>
<dd><p>Wrapper for overall counts to fold change conversion</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_lbl_ref_fn</strong> – filename of table with counts of mutations from reference condition</li>
<li><strong>data_lbl_sel_fn</strong> – filename of table with counts of mutations from selected condition</li>
<li><strong>info</strong> – dict with information of the experiment</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.data_lbl2data_fit_lite">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">data_lbl2data_fit_lite</code><span class="sig-paren">(</span><em>fits_pairs</em>, <em>prj_dh</em>, <em>data_lbl_dh</em>, <em>data_fit_dh</em>, <em>force=False</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.data_lbl2data_fit_lite" title="Permalink to this definition">¶</a></dt>
<dd><p>Short wrapper for conversion of mutation counts to fold changes</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>fits_pair</strong> – list with pair of selected and reference condition</li>
<li><strong>pj_dh</strong> – path to the protject directory</li>
<li><strong>data_lbl_dh</strong> – path to the diorectory containing data_lbl csv tables</li>
<li><strong>data_fit_dh</strong> – path to the directory of data_fit csvs</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.getNS">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">getNS</code><span class="sig-paren">(</span><em>data_all</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.getNS" title="Permalink to this definition">¶</a></dt>
<dd><p>This function separates the non-synonymous(N) and synonymous(S) mutations.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data_all</strong> – dataframe with frequency of mutations (<cite>data_lbl</cite>).</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_NS:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body">dataframe with non-synonymous(N) and synonymous(S) mutations.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.get_data_lbl_reps">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">get_data_lbl_reps</code><span class="sig-paren">(</span><em>data_lbl_fn</em>, <em>data_lbl_type</em>, <em>repli</em>, <em>info</em>, <em>data_fit=None</em>, <em>data_lbl_col='NiA_tran'</em>, <em>type_form='aas'</em>, <em>col_sep='.'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.get_data_lbl_reps" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets the replicates for a given filename of data_lbl</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_lbl_fn</strong> – filename of data_lbl</li>
<li><strong>data_lbl_type</strong> – codon level or amino acid level mutations</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.get_data_lbl_type">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">get_data_lbl_type</code><span class="sig-paren">(</span><em>data_fit</em>, <em>data_lbl_type</em>, <em>data_lbl_col='NiA_tran'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.get_data_lbl_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets the type of mutations in a given data_lbl table</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_fit</strong> – data_fit table</li>
<li><strong>data_lbl_type</strong> – type of mutations eg. codon level or amino acid level</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.make_GLM_norm">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">make_GLM_norm</code><span class="sig-paren">(</span><em>data_lbl_ref_fn</em>, <em>data_lbl_sel_fn</em>, <em>data_fit</em>, <em>info</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.make_GLM_norm" title="Permalink to this definition">¶</a></dt>
<dd><p>Wrapper for DESeq2 mediated GLM normalization</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_lbl_ref</strong> – pandas table with counts of mutations from reference condition</li>
<li><strong>data_lbl_sel</strong> – pandas table with counts of mutations from selected condition</li>
<li><strong>data_fit</strong> – pandas table with fold change values</li>
<li><strong>info</strong> – dict with information of the experiment</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.make_data_fit">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">make_data_fit</code><span class="sig-paren">(</span><em>data_lbl_ref_fn</em>, <em>data_lbl_sel_fn</em>, <em>info</em>, <em>data_lbl_col='NiA_tran'</em>, <em>type_form='aas'</em>, <em>test='ztest'</em>, <em>multitest='fdr_bh'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.make_data_fit" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimates fold changes</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_lbl_ref_fn</strong> – filename of the file (data_lbl) containing counts of mutations from the reference condition</li>
<li><strong>data_lbl_sel_fn</strong> – filename of the file (data_lbl) containing counts of mutations from the selection condition</li>
<li><strong>info</strong> – dict, information of the experiment</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.make_deseq2_annot">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">make_deseq2_annot</code><span class="sig-paren">(</span><em>unsel</em>, <em>sel</em>, <em>data_lbl_col</em>, <em>prj_dh</em>, <em>type_form='aas'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.make_deseq2_annot" title="Permalink to this definition">¶</a></dt>
<dd><p>Makes DESeq2 annotation file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>unsel</strong> – unselected condition</li>
<li><strong>sel</strong> – selected condtition</li>
<li><strong>prj_dh</strong> – path to the project directory</li>
<li><strong>data_lbl_col</strong> – column of the data_lbl pandas table to be used</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.make_deseq2_count">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">make_deseq2_count</code><span class="sig-paren">(</span><em>unsel</em>, <em>sel</em>, <em>data_deseq2_annot</em>, <em>data_lbl_col</em>, <em>prj_dh</em>, <em>type_form='aas'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.make_deseq2_count" title="Permalink to this definition">¶</a></dt>
<dd><p>Makes DESeq2 count file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>unsel</strong> – unselected condition</li>
<li><strong>sel</strong> – selected condtition</li>
<li><strong>prj_dh</strong> – path to the project directory</li>
<li><strong>data_lbl_col</strong> – column of the data_lbl pandas table to be used</li>
<li><strong>data_deseq2_annot</strong> – pandas table with annotation information</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.makemutids">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">makemutids</code><span class="sig-paren">(</span><em>data_lbl</em>, <em>refis</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.makemutids" title="Permalink to this definition">¶</a></dt>
<dd><p>This makes the mutation ids eg. M001T or A024T from mutation matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>data_lbl</strong> – dataframe with columns of reference (<cite>ref</cite>) and mutation (<cite>mut</cite>).</li>
<li><strong>refis</strong> – index of reference amino acids/ codons.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns mutids:</th><td class="field-body"><p class="first last">list of mutation ids.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.makemutids_fromprtseq">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">makemutids_fromprtseq</code><span class="sig-paren">(</span><em>prt_seq</em>, <em>muts=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.makemutids_fromprtseq" title="Permalink to this definition">¶</a></dt>
<dd><p>Make mutation IDs from a given protein sequence</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>prt_seq</strong> – string, protein sequence</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.mat_cds2mat_aas">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">mat_cds2mat_aas</code><span class="sig-paren">(</span><em>mat_cds</em>, <em>host</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.mat_cds2mat_aas" title="Permalink to this definition">¶</a></dt>
<dd><p>This converts mutation matrix of codon to that of amino acid.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>mat_cds</strong> – codon level mutation matrix.</li>
<li><strong>host</strong> – name of host organism for choosing codon table. [coli | yeast | sapiens].</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns mat_aas:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">amino acid level mutation matrix.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.mut_mat_cds2data_lbl">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">mut_mat_cds2data_lbl</code><span class="sig-paren">(</span><em>lbli</em>, <em>lbl_mat_mut_cds_fh</em>, <em>host</em>, <em>prj_dh</em>, <em>reflen</em>, <em>cctmr</em>, <em>Ni_cutoff</em>, <em>fsta_fh</em>, <em>clips=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.mut_mat_cds2data_lbl" title="Permalink to this definition">¶</a></dt>
<dd><p>This function converts mutation matrix (produced from .mat) file into data_lbl format.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>lbli</strong> – name of the sample.</li>
<li><strong>lbl_mat_mut_cds_fh</strong> – path to codon level mutation matrix (.mat_mut_cds).</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>:param host : name of host organism for choosing codon table. (coli | yeast | sapiens).
:param prj_dh: path to project directory.
:param reflen: nucleotide level length of the reference sequence.
:param cctmr: tuple with (1-based) boundaries of concatamers.
:param Ni_cutoff: threshold of depth per codon to be processed further.</p>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.mutids_converter">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">mutids_converter</code><span class="sig-paren">(</span><em>mutids</em>, <em>out</em>, <em>type_form</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.mutids_converter" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert Mutation IDs to different formats</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>mutids</strong> – list of mutation IDs</li>
<li><strong>out</strong> – format of the mutations</li>
<li><strong>type_form</strong> – amino acid level or codon level mutations</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.rescale_fitnessbysynonymous">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">rescale_fitnessbysynonymous</code><span class="sig-paren">(</span><em>data_fit</em>, <em>col_fit='FCA_norm'</em>, <em>col_fit_rescaled='FiA'</em>, <em>syn2nan=True</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.rescale_fitnessbysynonymous" title="Permalink to this definition">¶</a></dt>
<dd><p>Rescale fold changes by the fold change of synonymous mutations at that position</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data_fit</strong> – pandas table with fold change values</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.transform_data_lbl">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">transform_data_lbl</code><span class="sig-paren">(</span><em>prj_dh</em>, <em>transform_type</em>, <em>type_form='aas'</em>, <em>data_lbl_col='NiA_norm'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.transform_data_lbl" title="Permalink to this definition">¶</a></dt>
<dd><p>Transforamtion of counts of mutants in data_lbl table</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>prj_dh</strong> – path to the project directory</li>
<li><strong>transform_type</strong> – type of transformation log, plog, glog etc</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_lbl:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">data_lbl with transformed counts</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_mut_files.transform_data_lbl_deseq">
<code class="descclassname">dms2dfe.lib.io_mut_files.</code><code class="descname">transform_data_lbl_deseq</code><span class="sig-paren">(</span><em>prj_dh</em>, <em>transform_type</em>, <em>rscript_fh</em>, <em>type_form='aas'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_mut_files.transform_data_lbl_deseq" title="Permalink to this definition">¶</a></dt>
<dd><p>Transforamtion of counts of mutants in data_lbl table using DESeq2</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>prj_dh</strong> – path to the project directory</li>
<li><strong>transform_type</strong> – type of transformation rlog or VST</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Returns data_lbl:</th></tr>
<tr class="field-even field"><td>&#160;</td><td class="field-body"><p class="first last">data_lbl with transformed counts</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.plot_pdb"></span><div class="section" id="plot-pdb">
<h2><code class="docutils literal"><span class="pre">plot_pdb</span></code><a class="headerlink" href="#plot-pdb" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.plot_pdb.vector2bfactor">
<code class="descclassname">dms2dfe.lib.plot_pdb.</code><code class="descname">vector2bfactor</code><span class="sig-paren">(</span><em>vector</em>, <em>pdb_fh</em>, <em>pdb_clrd_fh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_pdb.vector2bfactor" title="Permalink to this definition">¶</a></dt>
<dd><p>Incorporates vector with values to the B-factor of PDB file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>vector</strong> – vector with values</li>
<li><strong>pdb_fh</strong> – path of input PDB file</li>
<li><strong>pdb_clrd_fh</strong> – path of output PDB file</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.variant_caller"></span><div class="section" id="id8">
<h2><code class="docutils literal"><span class="pre">io_data_files</span></code><a class="headerlink" href="#id8" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.variant_caller.get_mut_cds">
<code class="descclassname">dms2dfe.lib.variant_caller.</code><code class="descname">get_mut_cds</code><span class="sig-paren">(</span><em>qry_alind</em>, <em>fsta_seq</em>, <em>list_mut_cds_fh</em>, <em>Q_cutoff</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.variant_caller.get_mut_cds" title="Permalink to this definition">¶</a></dt>
<dd><p>This gets the mutations at codon level.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>qry_alind</strong> – aligned query sequnce.</li>
<li><strong>fsta_seq</strong> – sequence of reference DNA (str).</li>
<li><strong>list_mut_cds_fh</strong> – path to file with codon level list of mutations.</li>
<li><strong>Q_cutoff</strong> – cut off of average phred score per read.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.variant_caller.getusablesbams_list">
<code class="descclassname">dms2dfe.lib.variant_caller.</code><code class="descname">getusablesbams_list</code><span class="sig-paren">(</span><em>prj_dh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.variant_caller.getusablesbams_list" title="Permalink to this definition">¶</a></dt>
<dd><p>This detects sorted bam files that can be further processed for variant calling.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>prj_dh</strong> – path to project directory.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.variant_caller.is_qry_alind_useful">
<code class="descclassname">dms2dfe.lib.variant_caller.</code><code class="descname">is_qry_alind_useful</code><span class="sig-paren">(</span><em>qry_alind</em>, <em>Q_cutoff</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.variant_caller.is_qry_alind_useful" title="Permalink to this definition">¶</a></dt>
<dd><p>This detects whether a read passes a number of criterias eg. Phred quality score.</p>
<div class="highlight-text"><div class="highlight"><pre><span></span>cigar dode | descr | col index
M  BAM_CMATCH  0
I  BAM_CINS    1
D  BAM_CDEL    2
N  BAM_CREF_SKIP   3
S  BAM_CSOFT_CLIP  4
H  BAM_CHARD_CLIP  5
P  BAM_CPAD    6
=  BAM_CEQUAL  7
X  BAM_CDIFF   8
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>qry_alind</strong> – aligned query sequnce.</li>
<li><strong>Q_cutoff</strong> – cut off of average phred score per read.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.variant_caller.qual_chars2nums">
<code class="descclassname">dms2dfe.lib.variant_caller.</code><code class="descname">qual_chars2nums</code><span class="sig-paren">(</span><em>qual_chars</em>, <em>qual_format</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.variant_caller.qual_chars2nums" title="Permalink to this definition">¶</a></dt>
<dd><p>This converts the phred qualities from the ASCII format to numbers.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>qual_chars</strong> – phred score in ASCII format (str).</li>
<li><strong>qual_format</strong> – phred score in numbers (list).</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.variant_caller.sam2mutmat">
<code class="descclassname">dms2dfe.lib.variant_caller.</code><code class="descname">sam2mutmat</code><span class="sig-paren">(</span><em>sbam_fh</em>, <em>fsta_id</em>, <em>fsta_seqlen</em>, <em>fsta_seq</em>, <em>cds_ref</em>, <em>Q_cutoff</em>, <em>prj_dh</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.variant_caller.sam2mutmat" title="Permalink to this definition">¶</a></dt>
<dd><p>This converts sorted .bam file to codon level mutation matrix (.mat_mut_cds)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>sbam_fh</strong> – path to sorted bam file (str).</li>
<li><strong>fsta_id</strong> – name of reference (from .fasta file) (str).</li>
<li><strong>fsta_seqlen</strong> – length of reference sequnce (int).</li>
<li><strong>fsta_seq</strong> – reference sequnce (str).</li>
<li><strong>cds_ref</strong> – codons in refernce sequnce (list)</li>
<li><strong>Q_cutoff</strong> – cut off of average phred score per codon.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.plot_mut_data"></span><div class="section" id="plot-mut-data">
<h2><code class="docutils literal"><span class="pre">plot_mut_data</span></code><a class="headerlink" href="#plot-mut-data" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.annotate_pearsonr">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">annotate_pearsonr</code><span class="sig-paren">(</span><em>x</em>, <em>y</em>, <em>**kws</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.annotate_pearsonr" title="Permalink to this definition">¶</a></dt>
<dd><p>Annotates person’s r on a axes object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>x</strong> – location x</li>
<li><strong>y</strong> – location y</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.data2mut_matrix">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">data2mut_matrix</code><span class="sig-paren">(</span><em>data_fit</em>, <em>values_col</em>, <em>index_col</em>, <em>type_form</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.data2mut_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>This creates mutation matrix from input data (frequncies <cite>data_lbl</cite> or <cite>data_fit</cite>).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_fit</strong> – fitness data (<cite>data_fit</cite>).</li>
<li><strong>values_col</strong> – column name with values (str).</li>
<li><strong>index_col</strong> – column name with index (str).</li>
<li><strong>type_form</strong> – type of values [“aas” : amino acid | “cds” : codons].</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.data2sub_matrix">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">data2sub_matrix</code><span class="sig-paren">(</span><em>data_fit</em>, <em>values_col</em>, <em>index_col</em>, <em>type_form</em>, <em>aggfunc='mean'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.data2sub_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>This creates substitition matrix from input data (frequncies <cite>data_lbl</cite> or <cite>data_fit</cite>).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_fit</strong> – fitness data (<cite>data_fit</cite>).</li>
<li><strong>values_col</strong> – column name with values (str).</li>
<li><strong>index_col</strong> – column name with index (str).</li>
<li><strong>type_form</strong> – type of values [“aas” : amino acid | “cds” : codons].</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plot_corr_mat">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plot_corr_mat</code><span class="sig-paren">(</span><em>data_repli</em>, <em>plot_fh=None</em>, <em>xlim=(0</em>, <em>None)</em>, <em>ylim=(0</em>, <em>None)</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plot_corr_mat" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots a correlation matrix</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data_repli</strong> – pandas table with counts of replicates</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plot_cov">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plot_cov</code><span class="sig-paren">(</span><em>data_cov</em>, <em>data_lbl</em>, <em>plot_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plot_cov" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots the coverage (depth) of sequencing</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_cov</strong> – pandas dataframe with position wise coverage information</li>
<li><strong>data_lbl</strong> – pandas dataframe with counts of mutations</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plot_data_comparison_bar">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plot_data_comparison_bar</code><span class="sig-paren">(</span><em>data_comparison</em>, <em>plot_fh=None</em>, <em>index=[]</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plot_data_comparison_bar" title="Permalink to this definition">¶</a></dt>
<dd><p>This plots the proportion of mutants according to type of selection in effect.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data_comparison</strong> – input <cite>data_comparison</cite> (dataframe).</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plot_data_comparison_violin">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plot_data_comparison_violin</code><span class="sig-paren">(</span><em>data_comparison</em>, <em>plot_fh=None</em>, <em>stars=True</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plot_data_comparison_violin" title="Permalink to this definition">¶</a></dt>
<dd><p>plots verticle distributions i.e. violins from a data_comparison</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data_comparison</strong> – pandas table with two condition’s fold change values to be compared</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plot_data_fit_dfe">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plot_data_fit_dfe</code><span class="sig-paren">(</span><em>data_fit</em>, <em>norm_type=None</em>, <em>col_fit='FiA'</em>, <em>annot_X=True</em>, <em>annot_syn=False</em>, <em>xlabel='$F_{i}$'</em>, <em>axvspan_min=0</em>, <em>axvspan_max=0</em>, <em>plot_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plot_data_fit_dfe" title="Permalink to this definition">¶</a></dt>
<dd><p>This plots histogram of Distribution of Fitness Effects (DFE).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_fit</strong> – fitness data (<cite>data_fit</cite>).</li>
<li><strong>norm_type</strong> – Type of normalization across samples [wild: wrt wild type | syn : wrt synonymous mutations | none : fold change serves as fitness]</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plot_data_fit_scatter">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plot_data_fit_scatter</code><span class="sig-paren">(</span><em>data_fit</em>, <em>norm_type</em>, <em>Ni_cutoff</em>, <em>plot_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plot_data_fit_scatter" title="Permalink to this definition">¶</a></dt>
<dd><p>This plots scatter plot of frequecies of selected and unselected samples.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_fit</strong> – fitness data (<cite>data_fit</cite>).</li>
<li><strong>norm_type</strong> – Type of normalization across samples [wild: wrt wild type | syn : wrt synonymous mutations | none : fold change serves as fitness]</li>
<li><strong>Ni_cutoff</strong> – threshold of depth per codon to be processed further.</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plot_data_lbl_repli">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plot_data_lbl_repli</code><span class="sig-paren">(</span><em>prj_dh</em>, <em>lim_min=0</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plot_data_lbl_repli" title="Permalink to this definition">¶</a></dt>
<dd><p>This plots scatters of frequecies of replicates.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>prj_dh</strong> – path to project directory.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plot_sub_matrix">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plot_sub_matrix</code><span class="sig-paren">(</span><em>data_fit</em>, <em>type_form</em>, <em>col</em>, <em>cmap='coolwarm'</em>, <em>center=0</em>, <em>plot_fh=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plot_sub_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>This plots heatmap of fitness values.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_fit</strong> – input data (<cite>data_lbl</cite> or <cite>data_fit</cite>) (dataframe).</li>
<li><strong>type_form</strong> – type of values [“aas” : amino acid | “cds” : codons].</li>
<li><strong>col</strong> – eg. columns with values. col for data_fit</li>
<li><strong>cmap</strong> – name of colormap (str).</li>
<li><strong>center</strong> – center colormap to this value (int).</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plotacc">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plotacc</code><span class="sig-paren">(</span><em>data_feats</em>, <em>ax</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plotacc" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots surface accessibility</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_feats</strong> – pandas dataframe with position wise features</li>
<li><strong>ax</strong> – axes object</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plotss">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plotss</code><span class="sig-paren">(</span><em>data_feats</em>, <em>ax</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plotss" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots secondary structures</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>data_feats</strong> – pandas dataframe with position wise features</li>
<li><strong>ax</strong> – axes object</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.plot_mut_data.plotsspatches">
<code class="descclassname">dms2dfe.lib.plot_mut_data.</code><code class="descname">plotsspatches</code><span class="sig-paren">(</span><em>ssi</em>, <em>ssi_ini</em>, <em>aai</em>, <em>aai_ini</em>, <em>patches</em>, <em>patches_colors</em>, <em>ax</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.plot_mut_data.plotsspatches" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots secondary struncture
# H Alpha helix
# B Beta bridge
# E Strand
# G Helix-3
# I Helix-5
# T Turn
# S Bend</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>ssi</strong> – list with secondary sture elements</li>
<li><strong>ssi_ini</strong> – start index of sec. structures</li>
<li><strong>aai_ini</strong> – start amino acid index of sec. structures</li>
<li><strong>patches</strong> – shapes of sec. structures</li>
<li><strong>patches_colour</strong> – colours of sec. structures</li>
<li><strong>ax</strong> – axes object</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.io_nums"></span><div class="section" id="io-nums">
<h2><code class="docutils literal"><span class="pre">io_nums</span></code><a class="headerlink" href="#io-nums" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.io_nums.float2int">
<code class="descclassname">dms2dfe.lib.io_nums.</code><code class="descname">float2int</code><span class="sig-paren">(</span><em>x</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_nums.float2int" title="Permalink to this definition">¶</a></dt>
<dd><p>converts floats to int when only float() is not enough.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>x</strong> – float</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_nums.glog">
<code class="descclassname">dms2dfe.lib.io_nums.</code><code class="descname">glog</code><span class="sig-paren">(</span><em>x</em>, <em>l=2</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_nums.glog" title="Permalink to this definition">¶</a></dt>
<dd><p>Generalised logarithm</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>x</strong> – number</li>
<li><strong>p</strong> – number added befor logarithm</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_nums.is_numeric">
<code class="descclassname">dms2dfe.lib.io_nums.</code><code class="descname">is_numeric</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_nums.is_numeric" title="Permalink to this definition">¶</a></dt>
<dd><p>This detects whether an input object is numeric or not.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>obj</strong> – object to be tested.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_nums.plog">
<code class="descclassname">dms2dfe.lib.io_nums.</code><code class="descname">plog</code><span class="sig-paren">(</span><em>x</em>, <em>p=0.5</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_nums.plog" title="Permalink to this definition">¶</a></dt>
<dd><p>psudo-log</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>x</strong> – number</li>
<li><strong>p</strong> – number added befor logarithm</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_nums.str2num">
<code class="descclassname">dms2dfe.lib.io_nums.</code><code class="descname">str2num</code><span class="sig-paren">(</span><em>x</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_nums.str2num" title="Permalink to this definition">¶</a></dt>
<dd><p>This extracts numbers from strings. eg. 114 from M114R.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>x</strong> – string</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<span class="target" id="module-dms2dfe.lib.io_strs"></span><div class="section" id="io-strs">
<h2><code class="docutils literal"><span class="pre">io_strs</span></code><a class="headerlink" href="#io-strs" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="dms2dfe.lib.io_strs.convertstr2format">
<code class="descclassname">dms2dfe.lib.io_strs.</code><code class="descname">convertstr2format</code><span class="sig-paren">(</span><em>col</em>, <em>form</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_strs.convertstr2format" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert input string to input regex (<cite>form</cite>).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>col</strong> – input string.</li>
<li><strong>form</strong> – eg. for hdf5: <cite>“^[a-zA-Z_][a-zA-Z0-9_]*$”</cite></li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_strs.get_logger">
<code class="descclassname">dms2dfe.lib.io_strs.</code><code class="descname">get_logger</code><span class="sig-paren">(</span><em>argv=None</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_strs.get_logger" title="Permalink to this definition">¶</a></dt>
<dd><p>Initiates logging information in a pre-defined format.</p>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_strs.get_time">
<code class="descclassname">dms2dfe.lib.io_strs.</code><code class="descname">get_time</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_strs.get_time" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets current time in a form of a formated string. Used in logger function.</p>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_strs.isstrallowed">
<code class="descclassname">dms2dfe.lib.io_strs.</code><code class="descname">isstrallowed</code><span class="sig-paren">(</span><em>s</em>, <em>form</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_strs.isstrallowed" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks is input string conforms to input regex (<cite>form</cite>).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>s</strong> – input string.</li>
<li><strong>form</strong> – eg. for hdf5: <cite>“^[a-zA-Z_][a-zA-Z0-9_]*$”</cite></li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_strs.linebreaker">
<code class="descclassname">dms2dfe.lib.io_strs.</code><code class="descname">linebreaker</code><span class="sig-paren">(</span><em>l</em>, <em>break_pt=16</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_strs.linebreaker" title="Permalink to this definition">¶</a></dt>
<dd><p>used for adding labels in plots.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>l</strong> – list of strings</li>
<li><strong>break_pt</strong> – number, insert new line after this many letters</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_strs.make_pathable_string">
<code class="descclassname">dms2dfe.lib.io_strs.</code><code class="descname">make_pathable_string</code><span class="sig-paren">(</span><em>s</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_strs.make_pathable_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes symbols from a string to be compatible with directory structure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>s</strong> – string</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="dms2dfe.lib.io_strs.splitlabel">
<code class="descclassname">dms2dfe.lib.io_strs.</code><code class="descname">splitlabel</code><span class="sig-paren">(</span><em>label</em>, <em>splitby=' '</em>, <em>ctrl='__'</em><span class="sig-paren">)</span><a class="headerlink" href="#dms2dfe.lib.io_strs.splitlabel" title="Permalink to this definition">¶</a></dt>
<dd><p>used for adding labels in plots.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>label</strong> – string</li>
<li><strong>splitby</strong> – string split the label by this character/string</li>
<li><strong>ctrl</strong> – string, marker that denotes a control condition</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
        <br/>
        
<div id="sourcelink">
  <a href="../../_sources/v0.1.0_b161005/src/5api.rst.txt"
     rel="nofollow">Source</a>
</div>
      
    </p>
    <p>
        &copy; Copyright 2017, Rohan Dandage.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.6.5.<br/>
    </p>
  </div>
</footer>
  </body>
</html>